### DISCUSSION TITLE
Automate transaction import & monthly checklist

### DISCUSSION THREAD

**Discussion by ScottyRobinson** [serversideup/financial-freedom#131] [Ideas]
# 👉 Describe the problem
**Automated transactions** 
A lot of banks in the UK don't have an export feature - They do however allow you to connect banks to one another (weird) and in some cases have all banks under one app. Now, I am able to use that app to automatically export my transactions to a Google Sheet - My feature request would be that we're able to connect our Google Account (probably via oAuth) and have a command that syncs the Sheet to transactions in FF.

 **Monthly Checklist**
 This is where I am little ancient. I manually each month check to see if XYZ has been paid and I tick each bill that has been. I'd love to have something similar for 'regular payments' to know if all my KNOWN monthly/regular payments have been paid for the <time-range>.

# 🥰 Describe the "impact" on users?
Automation! Can only help users keep more in sync with their finances.

# 🏆 How to solve this problem
- New functionality to login / retrieve a Google Sheet
- Config the transaction sync and map each column to what a transaction needs on FF
- Sync via commands on a timely basis
- New functionality to determine which transactions are regular/scheduled 
- A new insight on whether 'regular' payments have been paid for the month

# 💯 How do we validate the problem is solved?
 - Users can sync their Google Sheet (in this case) which contain a live feed of their transactions.
 - Users are alerted/provided insight that their regular payments of the month have been paid without needing to dig for them.

**Comment by danpastori** [comment]
@ScottyRobinson This is a really, really fascinating idea! What I'd like to do is explore this option from a generic point of view that could possibly solve the same issue for a variety of users. 

## Making this work globally**
Google Sheets/Excel, my guess, would be the biggest budgeting product available. Even banks in the US, people export their CSVs to a spreadsheet and manage finances. If we could find a way to sync transactions from the spreadsheet, to the API, we could easily set up an automation process that'd work for a lot of users.

**Do you have all account transactions in one sheet?**
What it sounds like is you take all your accounts, merge them into a single app, then export all of them to a single Google Sheet. This is fine, I'll just have to find a way to map transaction account -> account. The rest of the mapping I have a simple GUI made that saves on behalf of an account.

**Generic Use**
I'd love to see if we could explore a way to do this type of process on a more "generic" level. Like if we could come up with a template, or standardized process that others could replicate, this would be much more useful to others in your same shoes. 

One thought which would be amazing, would be if we could set up a web-hook from your Google Sheet -> FF. Instead of a credential that pulls from your Google Drive, anytime a specific sheet is updated, it sends the data to FF. I could have brought light to the fact that I'm not super familiar with higher integrations in Google Sheets and maybe this isn't even possible, but this would open up the door to any other app using the same type of functionality. Would love to hear your thoughts!

## Rules for Transactions**
I have a basic UI to add rules for importing from a CSV in this PR: https://github.com/serversideup/financial-freedom/pull/129. When it's merged, I'll have another quick way to manage rules. Using this personally, I find it really convenient and automates 95% of my import. This will work the same for yours coming from any CSV type source.

## Scheduled/Recurring transactions
I've wanted this as well, moved this to an enhancement: https://github.com/serversideup/financial-freedom/issues/132

## Flag if Regular Payments have been paid
This could be interesting and might require a further discussion. Are you looking for something like the following scenario:
1. In my budget, I allocate $15.99 for Netflix each month.
2. I want to make sure that within the month of June I have a $15.99 charge for Netflix to my credit/debit or card.

(we might have to open up another discussion, this could easily be related to budgeting)

## Checklists, Notifications, etc.
We will be getting to this for sure. I want automated reminders, checklists, notifications for all accounts. The goal is to find the balance of keeping the user informed and involved in the budgeting process, without requiring too much input but also without doing everything for them. I found that when using other platforms, there were so many notifications, miss allocations, etc. That I just gave up.

Would love to hear your thoughts and if anyone else has something to contribute!
